sanic_routing/tree.py

```diff
@@ -57,186 +88,230 @@ def display(self) -> None:
-                            Line(
-                                f"{if_stmt} method in {route.methods}:",
-                                return_indent,
-                            ),
-                            Line(f"route_idx = {i}", return_indent + 1),
-                        ]
+                return_indent += 1
+
+            for group in sorted(self.groups, key=self._group_sorting):
+                group_bump = 0
+
+                # If the route had some requirements, let's make sure we check
+                # them in the source
+                if group.requirements:
+                    route_idx = "route_idx"
+                    self._inject_requirements(
+                        location, return_indent + group_bump, group
+                    )
+
+                # This is for any inline regex routes. It sould not include,
+                # path or path-like routes.
+                if group.regex:
+                    self._inject_regex(
+                        location, return_indent + group_bump, group
                     )
-                location.extend(
-                    [
-                        Line("else:", return_indent),
-                        Line("raise NoMethod", return_indent + 1),
-                    ]
+                    group_bump += 1
+
+                # Since routes are grouped, we need to know which to select
+                # Inside the compiled source, we keep track so we know which
+                # handler to assign this to
+                if route_idx == 0 and len(group.routes) > 1:
+                    route_idx = "route_idx"
+                    self._inject_method_check(
+                        location, return_indent + group_bump, group
+                    )
+
+                # The return.kingdom
+                self._inject_return(
+                    location, return_indent + group_bump, route_idx, group
                 )
-            routes = "regex_routes" if self.group.regex else "dynamic_routes"
-            route_return = (
-                "" if self.group.router.stacking else f"[{route_idx}]"
-            )
+
+        return src, delayed, final
+
+    def add_child(self, child: "Node") -> None:
+        self._children[child.part] = child
+
+    def _inject_param_check(self, location, indent, idx):
+        """
+        Try and cast relevant path segments.
+        """
+        unquote_start = "unquote(" if self.unquote else ""
+        unquote_end = ")" if self.unquote else ""
+        lines = [
+            Line("try:", indent),
+            Line(
+                f"basket['__matches__'][{idx}] = "
+                f"{unquote_start}{self.param.cast.__name__}(parts[{idx}])"
+                f"{unquote_end}",
+                indent + 1,
+            ),
+            Line("except ValueError:", indent),
+            Line("pass", indent + 1),
+            Line("else:", indent),
+        ]
+        self.base_indent += 1
+
+        location.extend(lines)
```

Comment:
Could use `location += lines`

Reply:
Generally I do. I opted for `extend` since there is such a large amount of `extend/append`. I think it makes the code easier to reason about than when it is constantly shifting between operators and methods.